File: velolabs_repos/lattis_android/app/src/main/java/com/lattis/ellipse/presentation/setting/IntPref.java
package com.lattis.ellipse.presentation.setting;

import android.content.SharedPreferences;

/**
 * Created by ssd3 on 7/24/17.
 */

public class IntPref extends AbstractPref<Integer> {

    public IntPref(SharedPreferences sharedPreferences, String settingKey, int defaultValue) {
        super(sharedPreferences, settingKey, defaultValue);
    }

    @Override
    public Integer getValue() {
        return getSharedPreferences().getInt(getSettingKey(),getDefaultValue());
    }

    @Override
    public void setValue(Integer value) {
        SharedPreferences.Editor editor = getSharedPreferences().edit();
        editor.putInt(getSettingKey(),value);
        editor.apply();
    }
}

File: velolabs_repos/lattis_android_v2/app/src/main/java/com/lattis/lattis/utils/settings/IntPref.kt
package com.lattis.lattis.utils.settings

import android.content.SharedPreferences

class IntPref(
    sharedPreferences: SharedPreferences,
    settingKey: String,
    defaultValue: Int
) : AbstractPref<Int?>(sharedPreferences, settingKey, defaultValue) {
    override var value: Int?
        get() = sharedPreferences.getInt(settingKey, defaultValue!!)
        set(value) {
            val editor = sharedPreferences.edit()
            editor.putInt(settingKey, value!!)
            editor.apply()
        }

}